### ISSUE TITLE
Slow responses and Crashing server (angular/webpack)

### ISSUE THREAD

**Issue by gkatsanos** [typicode/json-server#353]
Hello, I often get this after 5 minutes of starting the server (randomly)
```
Done

  Resources
  http://localhost:3000/cardname

  Home
  http://localhost:3000

  Type s + enter at any time to create a snapshot of the database
  Watching...


<--- Last few GCs --->

  184018 ms: Mark-sweep 1260.3 (1433.0) -> 1253.5 (1439.0) MB, 1690.8 / 1 ms [allocation failure] [GC in old space requested].
  185717 ms: Mark-sweep 1253.5 (1439.0) -> 1253.3 (1439.0) MB, 1698.7 / 0 ms [allocation failure] [GC in old space requested].
  187459 ms: Mark-sweep 1253.3 (1439.0) -> 1253.2 (1438.0) MB, 1742.4 / 0 ms [last resort gc].
  189181 ms: Mark-sweep 1253.2 (1438.0) -> 1253.2 (1438.0) MB, 1721.9 / 0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x3c90e59c9e31 <JS Object>
    1: _remove [/usr/local/lib/node_modules/json-server/node_modules/chokidar/index.js:517] [pc=0x25d1cb5201e1] (this=0x274a0b404519 <an FSWatcher with map 0x5d5a245d7b1>,directory=0xe0815dface1 <String[83]: /Users/gplus/Library/Logs/WebStorm2016.2/threadDumps-20160901-194804-WS-162.1628.41>,item=0xe0815dfad09 <String[15]: 20160903-171226>)
    2: /* anonymous */(aka /* anonymous */) [/usr/loca...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [/usr/local/bin/node]
 2: node::FatalException(v8::Isolate*, v8::Local<v8::Value>, v8::Local<v8::Message>) [/usr/local/bin/node]
 3: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [/usr/local/bin/node]
 4: v8::internal::Factory::NewFixedArray(int, v8::internal::PretenureFlag) [/usr/local/bin/node]
 5: v8::internal::OrderedHashTable<v8::internal::OrderedHashSet, v8::internal::JSSetIterator, 1>::Rehash(v8::internal::Handle<v8::internal::OrderedHashSet>, int) [/usr/local/bin/node]
 6: v8::internal::OrderedHashSet::Add(v8::internal::Handle<v8::internal::OrderedHashSet>, v8::internal::Handle<v8::internal::Object>) [/usr/local/bin/node]
 7: v8::internal::KeyAccumulator::AddStringKey(v8::internal::Handle<v8::internal::Object>, v8::internal::AddKeyConversion) [/usr/local/bin/node]
 8: v8::internal::KeyAccumulator::AddKey(v8::internal::Handle<v8::internal::Object>, v8::internal::AddKeyConversion) [/usr/local/bin/node]
 9: v8::internal::KeyAccumulator::AddKeys(v8::internal::Handle<v8::internal::FixedArray>, v8::internal::AddKeyConversion) [/usr/local/bin/node]
10: v8::internal::GetKeys_Internal(v8::internal::Isolate*, v8::internal::Handle<v8::internal::JSReceiver>, v8::internal::Handle<v8::internal::JSReceiver>, v8::internal::KeyCollectionType, v8::internal::PropertyFilter, v8::internal::KeyAccumulator*) [/usr/local/bin/node]
11: v8::internal::JSReceiver::GetKeys(v8::internal::Handle<v8::internal::JSReceiver>, v8::internal::KeyCollectionType, v8::internal::PropertyFilter, v8::internal::GetKeysConversion) [/usr/local/bin/node]
12: v8::internal::Builtin_ObjectKeys(int, v8::internal::Object**, v8::internal::Isolate*) [/usr/local/bin/node]
13: 0x25d1cb20961b
Abort trap: 6
```

I am running webpack-dev-server in parallel (port 5000)

Also, the overall response time is super slow.. 

Here is my db.json file : 

```json
17:15 $ cat db.json 
{
    "cardname": [
            {
                "id": "Infernoid Onuncu",
                "status":"success",
                "data":{
                    "name":"Infernoid Onuncu",
                    "text":"Cannot be Normal Summoned/Set. Must be Special Summoned (from your hand or Graveyard) by banishing 3 \"Infernoid\" monsters from your hand or Graveyard while the total Levels and Ranks of all Effect Monsters you control are 8 or lower, and cannot be Special Summoned by other ways. When this card is Special Summoned: You can destroy all other monsters on the field. Once per turn, during either player's turn, when a Spell/Trap Card or effect is activated: You can Tribute 1 monster; negate the activation, and if you do, banish that card.",
                    "card_type":"monster",
                    "type":"Fiend / Effect",
                    "family":"fire",
                    "atk":3000,
                    "def":3000,
                    "level":10,
                    "property":null
                }
            },
            {
                "id": "Infernoid Harmadik",
                "status":"success",
                "data":{
                    "name":"Infernoid Onuncu",
                    "text":"Cannot be Normal Summoned/Set. Must be Special Summoned (from your hand or Graveyard) by banishing 3 \"Infernoid\" monsters from your hand or Graveyard while the total Levels and Ranks of all Effect Monsters you control are 8 or lower, and cannot be Special Summoned by other ways. When this card is Special Summoned: You can destroy all other monsters on the field. Once per turn, during either player's turn, when a Spell/Trap Card or effect is activated: You can Tribute 1 monster; negate the activation, and if you do, banish that card.",
                    "card_type":"monster",
                    "type":"Water / Things",
                    "family":"fire",
                    "atk":3000,
                    "def":3000,
                    "level":10,
                    "property":null
                }
            }
    ]
}
```

**Comment by gkatsanos** [comment]
I changed the port from 3000 to a random one and now it works. probably conflicting with something else (no idea what though)

**Comment by typicode** [comment]
Thanks for the update :+1:
I'm closing the issue then.

It may be related to the way files were being watched. I've published `v0.9.0-beta.2` which should improve that. You can install it using `npm install -g json-server@next`.
